packages/editor/src/components/block-list/insertion-point.js

```diff
@@ -41,77 +38,47 @@ class BlockInsertionPoint extends Component {
 		} );
 	}
 
-	onClick() {
-		const { layout, rootClientId, index, ...props } = this.props;
-		props.insertDefaultBlock( { layout }, rootClientId, index );
-		props.startTyping();
-		this.onBlurInserter();
-		if ( props.onInsert ) {
-			this.props.onInsert();
-		}
-	}
-
 	render() {
 		const { isInserterFocused } = this.state;
-		const { showInsertionPoint, showInserter } = this.props;
+		const { showInsertionPoint, canShowInserter } = this.props;
 
 		return (
 			<div className="editor-block-list__insertion-point">
 				{ showInsertionPoint && <div className="editor-block-list__insertion-point-indicator" /> }
-				{ showInserter && (
-					<div className={ classnames( 'editor-block-list__insertion-point-inserter', { 'is-visible': isInserterFocused } ) }>
-						<IconButton
-							icon="insert"
-							className="editor-block-list__insertion-point-button"
-							onClick={ this.onClick }
-							label={ __( 'Insert block' ) }
-							onFocus={ this.onFocusInserter }
-							onBlur={ this.onBlurInserter }
-						/>
+				{ canShowInserter && (
+					<div
+						onFocus={ this.onFocusInserter }
+						onBlur={ this.onBlurInserter }
+						className={
+							classnames( 'editor-block-list__insertion-point-inserter', {
+								'is-visible': isInserterFocused,
+							} )
+						}
+					>
+						<Inserter />
 					</div>
 				) }
 			</div>
 		);
 	}
 }
-export default compose(
-	withSelect( ( select, { clientId, rootClientId, canShowInserter } ) => {
-		const {
-			canInsertBlockType,
-			getBlockIndex,
-			getBlockInsertionPoint,
-			getBlock,
-			isBlockInsertionPointVisible,
-			isTyping,
-		} = select( 'core/editor' );
-		const {
-			getDefaultBlockName,
-		} = select( 'core/blocks' );
-		const blockIndex = clientId ? getBlockIndex( clientId, rootClientId ) : -1;
```

Comment:
The `clientId` behavior was totally broken anyways, and unused. Became obvious as I was trying to implement it for the "before first block" by rendering as first child of the `BlockList`.